### DISCUSSION TITLE
ru/models/rtdetr/

### DISCUSSION THREAD

**Discussion by giscus** [ultralytics/ultralytics#20666] [Docs]
# ru/models/rtdetr/

Ознакомьтесь с продуктом Baidu RT-DETR, детектором объектов в реальном времени на базе Vision Transformer, который отличается высокой точностью и адаптируемой скоростью вывода. Узнайте больше с помощью Ultralytics.

https://docs.ultralytics.com/ru/models/rtdetr/

**Comment by ILIAHHne63** [comment]
How to train DETR model from scratch, I set parameter pretrained=False, and on first epoch I have mAP50=0.65.

**Reply by ILIAHHne63** [reply]
My code:

def main(data_path, model_name):
    model1 = RTDETR("rtdetr-l.pt")

    results = model1.train(
        data=data_path,
        epochs=500,
        device=[4, 5, 6, 7],
        pretrained=False,
        name=model_name,
    )

**Comment by UltralyticsAssistant** [comment]
👋 Hello, thank you for sharing information about Baidu RT-DETR and its integration with Ultralytics! 🚀 For those interested in further details, we recommend visiting the [Docs](https://docs.ultralytics.com/) for comprehensive guides, including [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples. Many frequently asked questions are already addressed there.

If this discussion is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us assist you more effectively.

If you have a custom training ❓ Question, please share as much detail as possible, including dataset samples and training logs, and review our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community in the way that suits you best:  
- For real-time chat, join [Discord](https://discord.com/invite/ultralytics) 🎧  
- For in-depth discussions, visit [Discourse](https://community.ultralytics.com/)  
- Or connect with others on our [Subreddit](https://reddit.com/r/Ultralytics)

## Upgrade

To ensure you are using the latest features and fixes, upgrade the `ultralytics` package (and all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml)) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

## Environments

YOLO runs smoothly in any of these up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM ([GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/))
- **Amazon** Deep Learning AMI ([AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/))
- **Docker Image** ([Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/)) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

---

This is an automated response to help guide you. An Ultralytics engineer will also review your discussion and assist you soon. 😊
